src/views/ProductDetail.vue

```diff
@@ -97,43 +69,148 @@
+    async applyFeature(feature: string, type: string) {
+      if(type === 'color') this.selectedColor = feature;
+      else if(type === 'size') this.selectedSize = feature
+      await this.updateVariant();
+    },
+    getFeatures() {
+      const features = {} as any
+      this.product.variants.map((variant: any) => {
+        const size = getFeature(variant.featureHierarchy, '1/SIZE/');
+        const color = getFeature(variant.featureHierarchy, '1/COLOR/');
+        if (!features[color]) features[color] = [size];
+        else if(!features[color].includes(size)) features[color].push(size);
+      });
+
+      Object.keys(features).forEach((color) => this.features[color] = sortSizes(features[color]))
+
+      this.selectedColor = Object.keys(this.features)[0];
+      this.selectedSize = this.features[this.selectedColor][0];
+    },
+    async updateVariant() {
+      let variant;
+      if(this.selectedColor || this.selectedSize) {
+        variant = this.product.variants.find((variant: any) => {
+          const hasSize = getFeature(variant.featureHierarchy, '1/SIZE/') === this.selectedSize;
+          const hasColor = getFeature(variant.featureHierarchy, '1/COLOR/') === this.selectedColor;
+          if (hasSize && hasColor) return variant;
+        })
+
+        if(!variant) {
+          this.selectedSize = this.features[this.selectedColor][0];
+          variant = this.product.variants.find((variant: any) => getFeature(variant.featureHierarchy, '1/SIZE/') === this.selectedSize)
+          showToast(translate("Selected variant not available"));
+        }
+      }
+      this.currentVariant = variant;
+      this.currentStoreInventory = this.otherStoresInventory = this.warehouseInventory = 0;
+      this.otherStoresInventoryDetails = []
+      await this.checkInventory(this.currentVariant.productId);
+    },
+    async checkInventory(productId: string) {
+      const resp: any = await StockService.checkInventory({
+        "filters": { "productId": productId },
+        "fieldsToSelect": ["productId", "atp", "facilityName", "facilityId", "facilityTypeId"],
+      });
+
+      if (resp.status === 200 && !hasError(resp) && resp.data.docs.length) {
+        resp.data.docs.filter((storeInventory: any) => {
+          if (storeInventory.facilityId === this.currentFacility.facilityId) {
+            if (storeInventory.facilityTypeId === 'WAREHOUSE') 
+              this.warehouseInventory = storeInventory.atp
+            return this.currentStoreInventory = storeInventory.atp
+          } else {
+            this.otherStoresInventoryDetails.push({facilityName: storeInventory.facilityName, stock: storeInventory.atp})
+            if (storeInventory.facilityTypeId === 'WAREHOUSE') 
+              this.warehouseInventory = storeInventory.atp
+            return this.otherStoresInventory += storeInventory.atp
+          }
+        })
+      } else {
+        this.currentStoreInventory = this.otherStoresInventory = this.warehouseInventory = 0
```

Comment:
In case of error we have initialized all the inventory to zero, but still we might have previous data in `otherStoresInventoryDetails`, so I think we should also clear its value.

Reply:
Updated sir